vulcan/lib/client/jsx/api/archimedes_api.js

```diff
@@ -0,0 +1,69 @@
+import {
+  checkStatus,
+  handleFetchSuccess,
+  handleFetchError,
+  headers
+} from 'etna-js/utils/fetch';
+
+import CwlSerializer from '../serializers/cwl';
+
+const archimedesPath = (endpoint) => `${CONFIG.archimedes_host}${endpoint}`;
```

Comment:
I think we'll want to remove most of this -- I'm thinking that vulcan communicates directly to vulcan, and that archimedes is not a standalone api.  The main motivating factors:

1.  Keeping as much of our core application logic in ruby, and not split between languages.
2.  Archimedes can be just about scheduling and sandboxing, without the added overhead of serialization and transports (removing a whole leg of work).
3.  Vulcan frontend can be self contained, only interacting with one core api (vulcan).

Reply:
My original concern about that was streaming large amounts of data from Archimedes -> Vulcan -> UI (instead of just Archimedes -> UI), but that was when I thought Archimedes would be providing the interface to the data store. If all that will be coming through Vulcan instead, I'll change all the UI routes to talk directly to Vulcan.

Reply:
Yessir!  Vulcan can stream.  In fact, _it might be possible for apache to stream it_.  For atleast this version, the storage will actually just be files on the file system locally.  We could point apache to that directory and serve them using X-Send just like metis.

Reply:
In fact, even if the storage backend changes, I imagine that it would make sense to always atleast buffer files somewhere locally, even if it's just in chunks, so that we could enable apache to do the streaming and not ruby.  It's possible to configure apache to handle streams from, say, a file system fifo or socket that the backend arranges on behalf of a more complex storage.  The more linux filesystem magic we can work so that we don't have streaming in our scripted languages the better!

Reply:
Right now it is fine for Vulcan to run cells via some process instead of a service - but at some point there will be a scheduler involved, and Vulcan will be submitting cells to this scheduler, with Archimedes executing via the scheduler job in some worker. How does all of this relate? It seems logical to put the scheduler into Polyphemus since we will want to schedule other things besides Vulcan running cells; the Archimedes worker could run anywhere, but must in the end deposit data in a place where the user can query it - this might mean keeping everything (except the scheduler) Vulcan-proximal as you suggest, so it's all easily streamable. I guess this would look like:
Now:  Vulcan calls sandboxed Archimedes instance to run cell, which writes results to local disk for Vulcan to stream
Then: Vulcan submits cell to scheduler, which runs it via a Vulcan worker, which writes results to local disk for Vulcan to stream

Reply:
There's tons of different iterations of what is possible.  For sure, our scheduler may have other work than vulcan jobs.  So basically, by having vulcan just make the invocation directly now, we don't bake any assumptions about how our job infrastructure will work in the end, and we have complete flexibilty about how to proceed, for sure.